### DISCUSSION TITLE
Suggestion: Allow hiding the link in the Reference Link widget

### DISCUSSION THREAD

**Discussion by DannyBen** [retypeapp/retype#105] [Ideas]
The fact that there is always a link in the `[!ref]` link widget prompted me to avoid using it in cases where I wanted to use it.

I would like to be able to hide it on a per-link basis (not global config - although if I had this, I would probably also use it since hiding all is better for me than showing all).

### Use cases:

- When the link is too long
- When it is an external link, especially if it is long - since it is unusual to show "https://..." to users

### Rationale:

It looks messy, and wraps in awkward ways, even when the link is not that long. The problem is even worse when there is an icon.

The problem is especially evident on narrow mobile screens. For example, a screenshot from the retype site itself:

![image](https://user-images.githubusercontent.com/2405099/126898223-4e9a4912-71d8-4d39-91c9-ae47b0b62cbc.png)

If this feature is considered, and the "how to implement" is open for debate, I can say I would expect it to be similar to the button widget, which also gives it room to grow further with more attributes.

For example:

```markdown
[!ref link=false text="Getting Started"](http://...)

# Or this (although not sure if this is great)
[!ref link=false Getting Started](http://...)

# Example for the future proofing with room for more attributes:
[!ref link=false text="Getting Started" variant="external"](http://...)
```

**Comment by fabriciomurta** [comment]
Thanks for the heads up. I was really thinking, while you want to remove the link from the component, it will mostly be reduced to a button. Well, but a different looking button... yet, if we do add the `variants` to that `ref` component, it will just return to the "button functionality of his"... except it stretches the page width and has a `clear` variant.

**Comment by geoffreymcgill** [comment]
Thanks for bringing this to our attention. I agree that this scenario looks messy and should be improved. 

We can likely do something where just the domain was printed and on mouse-over the full link could be rendered into a tooltip. We will experiment with some options.

**Reply by DannyBen** [reply]
Did this make it into the roadmap?

I like the "ref" component, as it makes it clear to the viewer that this is a link with a special meaning, but I encountered another annoyance with it, when it shows relative links with `../../`, which is very low level and technical. I would expect such links to be presented as absolute (`/something` instead of `../../something`)

<img width="535" alt="image" src="https://user-images.githubusercontent.com/2405099/226593411-899aae7b-7343-4a8f-a7ee-93f5da41d33c.png">
